---
title: "Class 19 Pertussis Mini Project"
author: "Dairabel"
format: pdf
---

# Web scraping 

Here I extract the CDC figures for Pertussis cases in USA:
https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html

```{r echo=FALSE}

cdc <- data.frame(
                                 Year = c(1922L,1923L,1924L,1925L,
                                          1926L,1927L,1928L,1929L,1930L,1931L,
                                          1932L,1933L,1934L,1935L,1936L,
                                          1937L,1938L,1939L,1940L,1941L,1942L,
                                          1943L,1944L,1945L,1946L,1947L,
                                          1948L,1949L,1950L,1951L,1952L,
                                          1953L,1954L,1955L,1956L,1957L,1958L,
                                          1959L,1960L,1961L,1962L,1963L,
                                          1964L,1965L,1966L,1967L,1968L,1969L,
                                          1970L,1971L,1972L,1973L,1974L,
                                          1975L,1976L,1977L,1978L,1979L,1980L,
                                          1981L,1982L,1983L,1984L,1985L,
                                          1986L,1987L,1988L,1989L,1990L,
                                          1991L,1992L,1993L,1994L,1995L,1996L,
                                          1997L,1998L,1999L,2000L,2001L,
                                          2002L,2003L,2004L,2005L,2006L,2007L,
                                          2008L,2009L,2010L,2011L,2012L,
                                          2013L,2014L,2015L,2016L,2017L,2018L,
                                          2019L),
         Cases = c(107473,164191,165418,152003,
                                          202210,181411,161799,197371,
                                          166914,172559,215343,179135,265269,
                                          180518,147237,214652,227319,103188,
                                          183866,222202,191383,191890,109873,
                                          133792,109860,156517,74715,69479,
                                          120718,68687,45030,37129,60886,
                                          62786,31732,28295,32148,40005,
                                          14809,11468,17749,17135,13005,6799,
                                          7717,9718,4810,3285,4249,3036,
                                          3287,1759,2402,1738,1010,2177,2063,
                                          1623,1730,1248,1895,2463,2276,
                                          3589,4195,2823,3450,4157,4570,
                                          2719,4083,6586,4617,5137,7796,6564,
                                          7405,7298,7867,7580,9771,11647,
                                          25827,25616,15632,10454,13278,
                                          16858,27550,18719,48277,28639,32971,
                                          20762,17972,18975,15609,18617)
       )
```

```{r}
library(ggplot2)
ggplot(cdc, aes(Year, Cases)) + 
  geom_line() + 
  geom_point() +
  geom_smooth() +
  labs(x="year", y="cases", title="Pertussis Cases in USA across the years")+ scale_y_continuous(labels=scales::label_comma())


```
in 1946 we got a vaccination! we should add it as a line.

```{r}
library(ggplot2)
ggplot(cdc, aes(Year, Cases)) + 
  geom_line() + 
  geom_point() +
  geom_smooth() +
  geom_vline(xintercept = 1946, color="red", linetype=2)+
  labs(x="year", y="cases", title="Pertussis Cases in USA across the years")+ scale_y_continuous(labels=scales::label_comma())


```

the US and other countries swtiched from original wP vaccine to new Ap Vaccine in 1996. lets add it to our plot.

```{r}
library(ggplot2)
ggplot(cdc, aes(Year, Cases)) + 
  geom_line() + 
  geom_point() +
  geom_smooth() +
  geom_vline(xintercept = 1946, color="red", linetype=2)+
  geom_vline(xintercept = 1996, color="blue", linetype=2)+
  labs(x="year", y="cases", title="Pertussis Cases in USA across the years")+ scale_y_continuous(labels=scales::label_comma())

```
```{r echo=FALSE}
chile <- data.frame(data.frame(
                             Year = c(2021L,2020L,2019L,2018L,2017L,2016L,
                                      2015L,2014L,2013L,2012L,2011L,2010L,
                                      2009L,2008L,2007L,2006L,2005L,2004L,
                                      2003L,2002L,2001L,2000L,1999L,1998L,
                                      1997L,1996L,1995L,1994L,1993L,1992L,
                                      1991L,1990L,1989L,1988L,1987L,1986L,
                                      1985L,1984L,1983L,1982L,1981L,1980L,
                                      1979L,1978L,1977L,1976L,1975L,1974L),
                            Cases = c(29L,62L,350L,682L,849L,752L,734L,
                                      1125L,1964L,5762L,2582L,794L,692L,969L,
                                      1030L,1288L,1213L,1059L,1106L,1171L,
                                      1611L,3554L,3178L,2482L,1457L,1094L,
                                      424L,208L,600L,264L,61L,59L,202L,
                                      224L,46L,38L,610L,1980L,127L,393L,
                                      2043L,2795L,446L,895L,10169L,10668L,
                                      2550L,12184L)
                     ))
```

```{r}
library(ggplot2)
ggplot(chile, aes(Year, Cases)) + 
  geom_line() + 
  geom_point() +
  geom_smooth() +
  geom_vline(xintercept = 1996, color="blue", linetype=2)+
  labs(x="year", y="cases", title="Pertussis Cases in Chile across the years")+ scale_y_continuous(labels=scales::label_comma())
```
```{r}
alldata=rbind(
  data.frame(chile, group="Chile"),
  data.frame(cdc, group="USA")
)
```

```{r}
library(ggplot2)
ggplot(alldata, aes(Year, Cases)) + 
  geom_line() + 
  geom_point() +
  geom_smooth() +
  facet_wrap(~group)+
  geom_vline(xintercept = 1996, color="blue", linetype=2)+
  labs(x="year", y="cases", title="Pertussis Cases across the years")+   scale_y_continuous(labels=scales::label_comma())
```


#Exploring CMI-PB data

Getting data as much as we can. getting RNAseq data. We wil use **jsonlite** package

```{r}
library(jsonlite)
```

```{r}
subject <-read_json("http://cmi-pb.org/api/subject", simplifyVector = TRUE)
head(subject)
```


>Q4. How many wP and aP subjects are there?

```{r}
table(subject$infancy_vac)
```
A: 49 with wP, 47 with aP

>Q5. How many Male and Female subjects/patients are in the dataset?



>Q6??. How many females non white individuals are there in the dataset?

```{r}
table(subject$race, subject$biological_sex)
```
A: 29 non white individuals. *counting the unknown*


##specimen table

```{r}
library(lubridate)
```

>Q7. Using this approach determine (i) the average age of wP individuals, (ii) the average age of aP individuals; and (iii) are they significantly different?







#specimen
##Joining multiple tables


```{r}
specimen <- read_json("http://cmi-pb.org/api/specimen", simplifyVector = TRUE)
head(specimen)
titer <- read_json("http://cmi-pb.org/api/ab_titer", simplifyVector = TRUE)
head(titer)
```
>Q9. Complete the code to join specimen and subject tables to make a new merged data frame containing all specimen records along with their associated subject details:

```{r}
library(dplyr)
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

>Q10. Now using the same procedure join meta with titer data so we can further analyze this data in terms of time of visit aP/wP, male/female etc.

```{r}
library(dplyr)
abdata <- inner_join(titer, meta)
dim(abdata)
head(abdata)
```
>Q11. How many specimens (i.e. entries in abdata) do we have for each isotype?

```{r}
table(abdata$isotype)
```
A: There are 6698 entries for IgE, 1413 for IgG, 6141 for IgG1, 6141 for IgG2, 6141 for IgG3, and 6141 for IgG4. We have six isotypes.

>Q12. What do you notice about the number of visit 8 specimens compared to other visits?

```{r}
print(table(abdata$visit))
```

A: We notice that it is the smallest at 80 compared to all others with four digits. Meaning that it is still going.

#4. Examine IgG1 Ab titer levels


```{r}
ig1 <- abdata %>%
  filter(isotype == "IgG1", visit!=8)
dim(ig1)
head(ig1)
```

>Q how many antigenrs are there?

```{r}
table(abdata$antigen)
```

> Q13. Complete the following code to make a summary boxplot of Ab titer levels (MFI) for all antigens:

```{r}
ggplot(ig1) +
  aes(MFI, antigen) +
  geom_boxplot() 
```

>Q14. What antigens show differences in the level of IgG1 antibody titers recognizing them over time? Why these and not others?

```{r}
ggplot(ig1) +
  aes(MFI, antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2)
```

A: the fim2/3 antigens shown differences in levels of IgG1, when we look at terminology it is the mixture of FIM2 and FIM3, the fim are fimbria. ALso we see the FHA which is Filamentous hemagglutinin. These are found on the exterior of the bacteria and involved with moving around/attaching and it makes sense that the vaccine would affect them and make them have higher levels of Ig1 antibody. 

peak around visit 5 or 6. th median is the bar in middle of the box.

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = TRUE) + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw()
```
```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2)
```
> Q15. Filter to pull out only two specific antigens for analysis and create a boxplot for each. You can chose any you like. Below I picked a “control” antigen (“Measles”, that is not in our vaccines) and a clear antigen of interest (“FIM2/3”, extra-cellular fimbriae proteins from B. pertussis that participate in substrate attachment).

```{r}
filter(ig1, antigen=="FHA") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = ) +
  facet_wrap(vars(visit)) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  labs(title="FHA antigen levels per visit")
```
second one is going to be FIM2/3

```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = ) +
  facet_wrap(vars(visit)) +
  theme_bw() +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  labs(title="FIM2/3 antigen levels per visit")
```
>Q16. What do you notice about these two antigens time courses and the FIM2/3 data in particular?

A: We notice that wP has higher average values from week 1-3 but then after that, the aP has higher average values than wP.

>Q17. Do you see any clear difference in aP vs. wP responses?

A: NO we dont see a big big difference, they are almost overlapping for each visit.

#5. Obtaining CMI-PB RNASeq data

```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE) 
```

we will se join again!

```{r}
ssrna <- inner_join(rna, meta)
dim(ssrna)
head(ssrna)
```
>Q18. Make a plot of the time course of gene expression for IGHG1 gene (i.e. a plot of visit vs. tpm).

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```
>Q19.: What do you notice about the expression of this gene (i.e. when is it at it’s maximum level)?

A: We notice that its peak is around visit 4.

>Q20. Does this pattern in time match the trend of antibody titer data? If not, why not?

No, its does not match the antibody titer, that one keeps increasing even after the expression is gone. 

```{r}
ggplot(ssrna) +
  aes(tpm, col=infancy_vac) +
  geom_histogram(alpha=0.5) +
  facet_wrap(vars(visit))
```

```{r}
ssrna %>%  
  filter(visit==4) %>% 
  ggplot() +
    aes(tpm, col=infancy_vac) + 
  geom_density() + 
    geom_rug() 
```

